msu/hooks/skills/skill.nut

```diff
@@ -101,10 +101,24 @@
 		if (!this.m.IsBaseValuesSaved)
 		{
 			this.b <- clone this.skill.m;
-			foreach (k, v in this.m)
+			local obj = this;
+			local tables = [];
+			while (obj.ClassName != "skill")
 			{
-				this.b[k] <- v;
+				tables.push(clone obj.m);
+				obj = obj[obj.SuperName];
 			}
+
+			// Iterate in reverse so that the values of slots with the same name in a parent
+			// are always taken from the bottom most child.
+			for (local i = tables.len() - 1; i >= 0; i--)
+			{
+				foreach (key, value in tables[i])
+				{
+					this.b[key] <- value;
+				}
+			}
```

Comment:
1. Clone is faster. Skill.nut has a lot of keys.
2. We would still need to iterate over tables in reverse.

Reply:
1. If you're sure clone is faster even if we then have to once again set the values of everything then fine.
2. Why? Since we're checking if it's already been declared, we don't need to anymore.

Reply:
1. We don't set the values for skill.nut again. Only for children. Thats why clone is way faster in this case.
2. Because the tables array is populated from child upwards. Do we have to iterate in reverse. There is no other way to populate the tables array.

Reply:
1) I see what you mean now, that's fine
2) I'm pretty sure the `in` check is sufficient, but because of point 1 this is irrelevant and your way is better anyway.